### PR TITLE
Feature/devsu 1487 use react in presentation sections

### PR DESCRIPTION
**Pull request by Anna-log7**
I refactored the slides and discussion sections to use React as well as added tests for their components.

### CODE REVIEW

**Review by kttkjl — CHANGES_REQUESTED** [review]
Running into problems when the user does not have enough permission to edit the page on slides, choose a name and press upload

1. The dialog comes up asking "Editing this report will ...etc ", but be completely unable to press cancel, due to React components being unmounted already

2. User gets a refreshed page and nothing happens

Shouldn't edit options be disabled if the user has no rights to it?

File: app/views/ReportView/components/Slides/components/UploadSlide/index.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      console.log(resp);
+      snackbar.enqueueSnackbar('Slide uploaded successfully', { variant: 'success' });
+      onUpload(resp);
+    } catch (err) {
+      snackbar.enqueueSnackbar(`Error uploading slide: ${err}`, { variant: 'error' });
+    }
+  }, [onUpload, slideName, snackbar, report, isSigned]);
+
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (Anna-log7) [reply app/views/ReportView/components/Slides/components/UploadSlide/index.tsx]: The `isSigned` check actually isn't needed here as the API request doesn't remove the signatures. I've removed it now.

Also fixed the select file styling

File: app/views/ReportView/components/Slides/components/UploadSlide/index.tsx [hunk 2]
```diff
@@ -66,7 +66,7 @@ const UploadSlide = ({
   }, [onUpload, slideName, snackbar, report]);
 
   return (
-    <>
+    <div className="upload">
       <TextField
         className="text-field-fix"
```
Comment (kttkjl) [inline app/views/ReportView/components/Slides/components/UploadSlide/index.tsx:71]: seems like the `.text-field-fix` causes margin problems 
image

**Review by kttkjl — APPROVED** [review]

File: app/views/ReportView/components/Slides/components/UploadSlide/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kttkjl) [reply app/views/ReportView/components/Slides/components/UploadSlide/index.tsx]: Permission problems not reproducible at this point ... might be GSC backend problems with keycloak, approving now

**Pull request merged**